File: fundamentals/factorial/factorial.ts
export {};

function factorial(n: number): number {
  if (n === 0) return 1;
  return n * factorial(n - 1);
}

function factorialIteration(n: number): number {
  let product = 1;
  for (let i = 2; i<= n; i++) {
    product *= i;
  }
  return product;
}

const num: number = 5;
console.log(`The Factorial of ${num} is: `, factorial(num));
console.log(`The Factorial using iteration for ${num} is: `, factorialIteration(num));

File: fundamentals/factorial/factorial.py
def factorial(n):
    if n == 0: return 1
    return n * factorial(n -1)

def factorial_iteration(n):
    product = 1
    for i in range(2, n + 1):
        product *= i
    return product

n = 5
print('The factorial of', n , 'is:', factorial(n))
print('The factorial of', n , 'with itegration is:', factorial_iteration(n))